client/blocks/credit-card-form/index.jsx

```diff
@@ -26,44 +25,46 @@ import support from 'lib/url/support';
 
 const wpcom = wpcomFactory.undocumented();
 
-const CreditCardForm = createReactClass( {
-	displayName: 'CreditCardForm',
+class CreditCardForm extends Component {
+	static displayName = 'CreditCardForm';
 
-	propTypes: {
+	static propTypes = {
 		apiParams: PropTypes.object,
 		createCardToken: PropTypes.func.isRequired,
 		initialValues: PropTypes.object,
 		recordFormSubmitEvent: PropTypes.func.isRequired,
 		saveStoredCard: PropTypes.func,
 		successCallback: PropTypes.func.isRequired,
 		showUsedForExistingPurchasesInfo: PropTypes.bool,
-	},
+	};
 
-	getInitialState() {
-		return {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
 			form: null,
 			formSubmitting: false,
 			notice: null,
 		};
-	},
-
-	_mounted: false,
-	fieldNames: [
-		'name',
-		'number',
-		'cvv',
-		'expirationDate',
-		'country',
-		'postalCode',
-		'streetNumber',
-		'address1',
-		'address2',
-		'phoneNumber',
-		'streetNumber',
-		'city',
-		'state',
-		'document',
-	],
+
+		this._mounted = false;
```

Comment:
alternatively we can use the `ref` property on the top-level React component (the `<form>`) and React will call it for us. on `mount` it will pass the DOM node as the reference and on `unmount` it will pass `null`

```js
storeForm = ref => this.form = ref;

…

doSomething = () => {
	if ( ! this.form ) { return; } // not mounted anymore

	…
}

…

<form ref={ this.storeForm } />
```